### PR TITLE
Simplified entity sign methods (fixes #161).

### LINKED ISSUE #161: User::create can come with signing
It seems like [User::create](https://github.com/radicle-dev/radicle-link/blob/master/librad/src/meta/user.rs#L162) could come signed with its own `SecretKey`. This would make it easier to bootstrap a user without needing a resolver.

### PR DESCRIPTION
**Pull request by massimiliano-mantione**
See #161...

### CODE REVIEW

File: librad/src/meta/entity.rs [hunk 1]
```diff
@@ -420,6 +420,50 @@ where
...
+    ///
+    /// - the entity has not been already signed using this same key
+    /// - this key is owned by the provided user
+    pub fn sign_by_user(&mut self, key: &SecretKey, user: &User<Verified>) -> Result<(), Error> {
+        let public_key = key.public();
+        if self.signatures().contains_key(&public_key) {
+            return Err(Error::SignatureAlreadyPresent(public_key));
+        }
+        if !user.has_key(&public_key) {
+            return Err(Error::UserKeyNotPresent(user.urn(), public_key));
```
Comment (FintanH) [inline librad/src/meta/entity.rs:457]: I would say this error is _sort_ of misleading. It's more the case that the `SecretKey`/`PublicKey` is not owned by this `User`. What do you think?

Comment (massimiliano-mantione) [reply librad/src/meta/entity.rs:457]: For now I was just consistent with what the original, generic `sign` method does: if you provide a signatory that is a `Signatory::User` and the user does not have the key, that is the error.
So, if we change it, we should change it in both places and keep it consistent.
Changing it means just renaming it, and I am definitely open to better error names...

Comment (FintanH) [reply librad/src/meta/entity.rs:457]: Let's keep it for now :+1:

**Comment by FintanH** [comment]
I'll let you do the honours :bow:

**Pull request merged**